Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject/DownloadObject.cs

```diff
@@ -144,17 +144,15 @@ private static bool FileLocked(int fileLockedRetries, string fullPath, Cancellat
     {
         try
         {
-            var retries = 0;
-
-            while (retries <= fileLockedRetries)
+            for(var i = 0; i <= fileLockedRetries;)
             {
                 cancellationToken.ThrowIfCancellationRequested();
                 using FileStream inputStream = File.Open(fullPath, FileMode.Open, FileAccess.Read, FileShare.None);
                 if (inputStream.Length > 0)
                     return false;
                 else
                 {
-                    retries++;
+                    i++;
```

Comment:
Incrementation of iteration variable can be moved to the `for`-parameters, since iteration variable will be incremented each time unless `inputStream.Length > 0` and if that is true then method will return. So `for (var i = 0; i <= fileLockedRetries; i++)`.